Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -1509,6 +1545,138 @@ DiscusMultipleScatteringCorrection::scatter(const int nScatters, Kernel::PseudoR
   return {true, weights};
 }
 
+/*
+ * Construct a hexahedron shape extending from the detector's front face across the collimator openning window toward
+ the sample with a legth as twice the distance from sample to detector.
+ */
+std::shared_ptr<Geometry::CSGObject> DiscusMultipleScatteringCorrection::createCollimatorHexahedronShape(
+    const Kernel::V3D &samplePos, const Mantid::Geometry::DetectorInfo &detectorInfo, const int64_t &histogramIndex) {
+  const auto shape = detectorInfo.detector(histogramIndex).shape();
+  try {
+    shape->shapeInfo();
+  } catch (std::exception &) {
+    return nullptr;
+  }
+
+  const auto &detectorId = detectorInfo.detector(histogramIndex).getID();
+  const auto &detectorAbsolPos = detectorInfo.position(detectorInfo.indexOf(detectorId));
+  const auto cuboidGeometry = shape->shapeInfo().cuboidGeometry();
+
+  // Positions of the detector's front face
+  const auto &detLeftFrontBottomPos = detectorAbsolPos + cuboidGeometry.leftFrontBottom;
+  const auto &detLeftFrontTopPos = detectorAbsolPos + cuboidGeometry.leftFrontTop;
+  const auto &detRightFrontBottomPos = detectorAbsolPos + cuboidGeometry.rightFrontBottom;
+  const auto detRightFrontTopPos = detLeftFrontTopPos + detRightFrontBottomPos - detLeftFrontBottomPos;
+
+  const auto detCentorPos =
```

Comment:
Possible typo?
```suggestion
  const auto detCenterPos =
```
I'm assuming this not always be the same as `detectorAbsolPos` which is the center of the 3D cuboid - whereas this is the center of the 2D cross-section of the detector as viewed from the sample?

Reply:
That is correct. This is the centre of the 2D cross section as viewed from the sample.